### PR TITLE
fix/remove jquery usage on components

### PR DESCRIPTION
**Pull request by JeremyDec**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #MANAGER-8558, MANAGER-8560
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch 
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally 
- [ ] ~~Ticket reference is mentioned in linked commits (internal only)~~
- [x] Breaking change is mentioned in relevant commits

### Description 

Remove jQuery dependency for the following components:
- `@ovh-ux/ng-ovh-actions-menu`: Replace jQuery methods by vanillajs =>introduce breaking change due to classList usage.
- `@ovh-ux/ng-ovh-contracts`
- `@ovh-ux/ng-ovh-responsive-popover`: fix class directive import + replace of jQuery usage by vanilla js (introduce breaking change)

### CODE REVIEW

File: packages/components/ng-ovh-responsive-popover/src/class/directive.js [hunk 1]
```diff
@@ -1,58 +0,0 @@
...
-  return {
-    restrict: 'C',
-    link(scope, iElement) {
-      const mobileMediaQuery = responsivePopover.getMobileMediaQuery();
-      let isMobileDetected = false;
-      let mediaQueryHandler = false;
-      let scrollPosition;
-      const resetFixedBody = function(scrollTo) {
-        $('body')
-          .removeClass('responsive-popover-fixed-body')
```
Comment (antleblanc) [inline packages/components/ng-ovh-responsive-popover/src/class/directive.js:25]: Can find this className into following file:
- packages/components/ng-ovh-responsive-popover/src/index.less

File: packages/components/ng-ovh-responsive-popover/src/class/directive.js [hunk 2]
```diff
@@ -1,58 +0,0 @@
...
-      };
-
-      // on method returns a function that remove the media query listener
-      // @see https://github.com/AnalogJ/matchmedia-ng/blob/master/matchmedia-ng.js#L75
-      mediaQueryHandler = matchmedia.on(mobileMediaQuery, (mediaQueryList) => {
-        isMobileDetected = mediaQueryList.matches;
-
-        if (isMobileDetected) {
-          scrollPosition = $('body').scrollTop();
-          iElement.addClass('popover-full-screen');
```
Comment (antleblanc) [inline packages/components/ng-ovh-responsive-popover/src/class/directive.js:37]: Can find this className into following file:
- packages/components/ng-ovh-responsive-popover/src/index.less

Also able to find the same className across multiple location:
- packages/components/ng-ovh-actions-menu/src/index.less
- packages/manager/apps/telecom/src/app/common/less/mixins.less
- packages/manager/apps/telecom/src/components/telecom/telephony/group/number/feature/conference/conference.less
- packages/manager/apps/telecom/src/components/telecom/telephony/group/number/number.less
- packages/manager/apps/telecom/src/components/telecom/telephony/scheduler/scheduler.less
- packages/manager/apps/telecom/src/components/telecom/telephony/service/choice-popover/choice-popover.less
- packages/manager/apps/telecom/src/components/telecom/telephony/timeCondition/time-condition.less

Don't know if we are relying on the same CSS rules.

File: packages/components/ng-ovh-responsive-popover/src/class/directive.js [hunk 3]
```diff
@@ -1,58 +0,0 @@
...
- *  @description
- *  This directive manage the way the popover is displayed. This uses the configuration setted into
- *  the `responsivePopoverProvider` to detect if the popover needs to be displayed on full screen
- *  or to be displayed normally.
- *  It is automatically setted by `responsivePopover` directive.
- *
- *  **Note:** when ui-bootstrap version >= 2.0 will be used, the only thing to do should be to add
- *  a custom class with the 'popover-class' option of the uibPopover directive. To be tested.
- */
-export default /* @ngInject */ function(matchmedia, responsivePopover) {
```
Comment (antleblanc) [inline packages/components/ng-ovh-responsive-popover/src/class/directive.js:15]: **question:** Does the following dependency is still needed `matchmedia-ng`?

File: packages/components/ng-ovh-contracts/package.json [hunk 4]
```diff
@@ -45,7 +45,6 @@
     "angular": "^1.6.10",
     "angular-translate": "^2.17.0",
     "angular-ui-bootstrap": "~1.3.3",
-    "bootstrap": "^4.0.0",
-    "jquery": "^2.1.3"
```
Comment (antleblanc) [inline packages/components/ng-ovh-contracts/package.json:49]: Good catch! Doesn't seems to be used from here.

**Review by antleblanc — CHANGES_REQUESTED** [review]

File: packages/components/ng-ovh-responsive-popover/src/class/directive.js (hunk 2 above)
Comment (JeremyDec) [reply packages/components/ng-ovh-responsive-popover/src/class/directive.js:37]: The directive in itself, isn't integrated in main angular module of the component: 

- It is define, and declared in its own module: https://github.com/ovh/manager/blob/master/packages/components/ng-ovh-responsive-popover/src/class/ovh-angular-responsive-popover-class.js, which is then exported.
.- But no import of this exported module: https://github.com/ovh/manager/blob/master/packages/components/ng-ovh-responsive-popover/src/index.js

However the less file can be clean to remove the unexisting class

Comment (JeremyDec) [reply packages/components/ng-ovh-responsive-popover/src/class/directive.js:37]: In fact I did delete this directive regardless of these usecases, as the module doesn't include it. But in fact there's a bug in manager telecom due to the absence of this directive. I'll change the code and try to replace jQuery by vanilla js.

Thx for the point

File: packages/components/ng-ovh-responsive-popover/src/class/directive.js (hunk 3 above)
Comment (JeremyDec) [reply packages/components/ng-ovh-responsive-popover/src/class/directive.js]: Seems still use by `$onDestroy` hook to check for mobile/desktop usage

File: packages/components/ng-ovh-actions-menu/src/directive.js [hunk 5]
```diff
@@ -107,12 +107,10 @@ export default function() {
...
         if (!focusHelper.length) {
-          focusHelper = $('<a></a>');
-          focusHelper.attr({
-            href: '#',
-            class: 'focus-helper',
-            onClick: 'return false;',
-          });
+          focusHelper = document.createElement('a');
+          focusHelper.setAttribute('href', '#');
+          focusHelper.classList.add('focus-helper');
```
[the remark this answers is not in the document]
Comment (JeremyDec) [reply packages/components/ng-ovh-actions-menu/src/directive.js:112]: Good catch, thx

**Review by antleblanc — DISMISSED** [review]

**Review by antleblanc — DISMISSED** [review]

File: packages/components/ng-ovh-actions-menu/src/directive.js [hunk 6]
```diff
@@ -121,26 +119,32 @@ export default function() {
 
         // manage focus loop for current page
         visiblePage.on('keydown', (event) => {
+          const firstLinkElement = visiblePage
```
Comment (marie-j) [inline packages/components/ng-ovh-actions-menu/src/directive.js:122]: Are we sure we always have a result ? (as we now take first array element)

File: packages/components/ng-ovh-responsive-popover/package.json [hunk 7]
```diff
@@ -38,8 +38,7 @@
     "set-value": "^2.0.1"
   },
   "dependencies": {
-    "@ovh-ux/translate-async-loader": "^1.0.8",
-    "jquery": "^2.1.3"
+    "@ovh-ux/translate-async-loader": "^1.0.8"
```
Comment (marie-j) [inline packages/components/ng-ovh-responsive-popover/package.json:41]: I notice that yarn.lock is not updated , jquery used elsewhere ?

File: packages/components/ng-ovh-actions-menu/src/directive.js (hunk 6 above)
Comment (JeremyDec) [reply packages/components/ng-ovh-actions-menu/src/directive.js:122]: In fact here, the `first()[0]` is here to get the element targeted HTML content, of the JSON returned by `find` + `first`, which is required to work with `event.target.matches`.

The jQuery method `is()` is able to work directly with the JSON return by `find` and `first`.

File: packages/components/ng-ovh-responsive-popover/package.json (hunk 7 above)
Comment (JeremyDec) [reply packages/components/ng-ovh-responsive-popover/package.json:41]: Yes ! jQuery still used by other packages

**Review by marie-j — DISMISSED** [review]

**Comment by JeremyDec** [comment]
Rebase on `develop` > changes on `ng-ovh-actions-menu` becomes useless.

**Review by anooparveti — APPROVED** [review]

**Review by marie-j — DISMISSED** [review]

**Comment by JeremyDec** [comment]
Rebase on `master`

**Review by anooparveti — DISMISSED** [review]

**Review by antleblanc — DISMISSED** [review]

**Review by marie-j — DISMISSED** [review]

**Review by antleblanc — APPROVED** [review]

**Pull request merged**